### PR TITLE
Add component URL to license report

### PR DESCRIPTION
**Pull request by danhermann**
Outputs component URL to report and validates that all components have a non-empty URL

### CODE REVIEW

File: tools/dependencies-report/src/main/java/org/logstash/dependencies/ReportGenerator.java [hunk 1]
```diff
@@ -45,32 +45,33 @@ public boolean generateReport(
...
-        Map<String, String> licenseMapping = new HashMap<>();
+        Map<String, String[]> licenseMapping = new HashMap<>();
         readLicenseMapping(licenseMappingStream, licenseMapping);
         List<String> acceptableLicenses = new ArrayList<>();
         readAcceptableLicenses(acceptableLicensesStream, acceptableLicenses);
         for (Dependency dependency : dependencies) {
             String nameAndVersion = dependency.name + ":" + dependency.version;
             if (licenseMapping.containsKey(nameAndVersion)) {
-                String mappedLicense = licenseMapping.get(nameAndVersion);
+                String[] mappedLicense = licenseMapping.get(nameAndVersion);
```
Comment (andrewvc) [inline tools/dependencies-report/src/main/java/org/logstash/dependencies/ReportGenerator.java]: Can we avoid `string[]` , while they are compact they make it hard to read the code. Can we make a value type instead? I know its more code, but I think it's worth it. We don't need to use accessors, just an object with public properties is fine.

File: tools/dependencies-report/src/test/resources/expectedOutput.txt [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
-jar-dependencies,0.3.10,Apache-2.0
-jar-dependencies,0.3.11,MIT
-jar-dependencies,0.3.12,MIT
-jruby-openssl,0.9.20,Apache-2.0
-jruby-openssl,0.9.21,Apache-2.0
-jruby-readline,1.1.1,Apache-2.0
-json-parser,,Apache-2.0
-junit:junit,4.12,Apache-2.0
-org.codehaus.janino:commons-compiler,3.0.8,BSD-3-Clause-Attribution
+dependencyName,dependencyVersion,url,license
```
Comment (andrewvc) [inline tools/dependencies-report/src/test/resources/expectedOutput.txt]: I'm realizing, while we're doing this, we should follow https://github.com/elastic/logstash/issues/8725 exactly and include the revision column, even though we'll never write to it. We should also use the correct column headers at the top there.

If you think that belongs in a separate PR, I'm fine with that as well.

Comment (danhermann) [reply tools/dependencies-report/src/test/resources/expectedOutput.txt]: I do think a separate PR would be better because that will involve lots of change to the unit tests.
